internal/oci/container_linux.go

```diff
@@ -104,3 +107,25 @@ func getPidStatDataFromFile(file string) (string, string, error) { //nolint:gocr
 
 	return string(statFields[stateFieldIndex]), string(statFields[startTimeFieldIndex]), nil
 }
+
+// SetRuntimeUser sets the runtime user for the container.
+func (c *Container) SetRuntimeUser(spec *specs.Spec) {
+	runtimeSpec := spec
+	if runtimeSpec.Process == nil {
+		logrus.Infof("Container %s has no process spec", c.ID())
+		return
+	}
+	user := runtimeSpec.Process.User
+	supplementalGroups := make([]int64, 0, len(user.AdditionalGids))
+	for _, gid := range user.AdditionalGids {
+		supplementalGroups = append(supplementalGroups, int64(gid))
```

Comment:
Since you know the exact size, a for-loop with an index would be better suited here. Avoid wasteful append.

Reply:
This is required to address to avoid out-of-range errors: https://github.com/cri-o/cri-o/actions/runs/9650477475/job/26616238838?pr=8268

Reply:
This code didn't appease CI:
```go
    user := runtimeSpec.Process.User
    additionalGids := user.AdditionalGids
    supplementalGroups := make([]int64, len(additionalGids))
   
    if len(additionalGids) > 0 {
           for idx := range additionalGids {
        supplementalGroups[idx] = int64(additionalGids[idx])
        }
    }
 
```

Reply:
Let me stick to the same logic as the [containerd](https://github.com/containerd/containerd/pull/9737/files#diff-4d3a2175fba09829ea97479a3b17f76cd4e57ec6409cc3e5177c327704a2ca3bR42) one.

Reply:
Looks like I need to preallocate `supplementalGroups`
```sh
  Running [/home/runner/golangci-lint-1.58.0-linux-amd64/golangci-lint run  --timeout=10m] in [/home/runner/work/cri-o/cri-o] ...
  Received 1619092 of 1619092 (100.0%), 1.5 MBs/sec
  Error: internal/oci/container_linux.go:118:2: Consider pre-allocating `supplementalGroups` (prealloc)
  	var supplementalGroups []int64
  	^
  
  Error: issues found

```

Reply:
Sure. Since the list might be empty. You'd need a guard somewhere to check if there is anything in the list, as an empty list will have nothing at index 0, indeed.